spydrnet_physical/util/tests/test_ConnectPointList.py

```diff
@@ -0,0 +1,219 @@
+		#self.assertEqual(pt.level,"same")
+
+	def test_reset_level(self):
+		'''This test checks whether the connection reset to the same level'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		self.pts.reset_level(pt)
+		self.assertEqual(pt.level,"same")
+
+	def test_hold_cursor(self):
+		'''This test checks whether the cursor holds its position when a connection is made in any coordonate'''
+		self.pts.cursor = (2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.assertEqual(self.pts.cursor,(2,3))
+
+	def test_release_cursor(self):
+		'''This test checks whether the cursor also moves its position when a connection is made in any coordonate'''
+		self.pts.set_cursor(2,3)
+		self.pts.hold_cursor()
+		self.pts.move_x(3)
+		self.pts.release_cursor()
+		self.pts.move_x(2).move_y(4)
+		self.assertEqual(self.pts.cursor,(4,7))
+
+	def test_flip_V(self):
+		'''This test checks whether the connection is correctly flipped vertically'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("V")
+		self.assertEqual(point.connection,(2,-2,2,-3))
+
+	def test_flip_H(self):
+		'''This test checks whether the connection is correctly flipped horizontally'''
+		pt = ConnectPoint(2, 2, 2, 3,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_from_point((2,2))
+		self.assertEqual(point.connection,(2,2,2,3))
+		pts.flip("H")
+		self.assertEqual(point.connection,(-2,2,-2,3))
+
+	def test_sample_connections(self):
+		'''This test checks whether the connection is sampled correctly equal to the max distance defined '''
+		pt = ConnectPoint(2, 2, 5, 2,level = "up")
+		pts = ConnectPointList(5,10,pt)
+		point = pts.search_to_point((5,2))
+		self.assertEqual(point.connection,(2,2,5,2))
+		pts.sample_connections(max_distance=1)
+		self.assertEqual(point.connection,(4,2,5,2))
+
+	def test_merge(self):
+		'''This test cheks whether two connect point lists are merged correclty'''
+		pts1 = ConnectPointList(5, 5)
+		pts1.cursor = (2,2)
+		pts1.move_x()
+		pts2 = ConnectPointList(5,5)
+		pts2.cursor = (2,3)
+		pts2.move_y()
+		pts1.merge(pts2)
+		point = pts1.search_from_point((2,3))
+		self.assertEqual(point.connection, (2, 3, 2, 4))
+
+	def test_scale(self):
+		'''This test checks whether the connect point list is scaled correctly according to the scale value and anchor coordinates defined '''
+		self.pts.cursor = (1,1)
+		self.pts.move_x(2)
+		self.pts.scale(scale=3,anchor=(0,0))
+		point = self.pts.search_from_point((3,3))
+		self.assertEqual(point.connection,(3,3, 9,3))
+
+	def test_translate(self):
+		'''This test checks whether the connect point list is translated correctly according the to the x and y coordinates defined '''
+		self.pts.cursor = (1,1)
+		self.pts.move_x(2)
+		self.pts.translate(2,0)
+		point = self.pts.search_from_point((3,1))
+		self.assertEqual(point.connection,(3,1, 5,1))
+
+	def test_rotate(self):
```

Comment:
Try all 4 roate option, also make sure it raises error when wring value is given

Reply:
In this method, positive angles are working correctly but when we give negative angle (-90,-180) ,it does not rotate and returns the original points. The negative angles are not implemented in the _rotate_point method in ConnectPoint which is used in the rotate method.

Reply:
Can we add that support as well, it simply means if the number is negative then perform `360-number`